apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableServiceApiClientPartial.cs

```diff
@@ -85,10 +89,60 @@ public partial class BigtableServiceApiSettings
                 retryFilter: IdempotentRetryFilter
             );
 
+        // TODO: Validate range, make sure this default makes sense
+        /// <summary>
+        /// 
+        /// </summary>
+        public int MaxChannels { get; set; } = 20;
+
+        // TODO: Validate range, make sure this default makes sense
+        /// <summary>
+        /// 
+        /// </summary>
+        public int PreferredMaxStreamsPerChannel { get; set; } = 4;
+
         partial void OnCopy(BigtableServiceApiSettings existing)
         {
             MutateRowsRetrySettings = existing.MutateRowsRetrySettings;
             ReadRowsRetrySettings = existing.ReadRowsRetrySettings;
+            MaxChannels = existing.MaxChannels;
+            PreferredMaxStreamsPerChannel = existing.PreferredMaxStreamsPerChannel;
+        }
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public static class BigtableServiceApiSettingsExtensions
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static IEnumerable<ChannelOption> CreateChannelOptions(this BigtableServiceApiSettings settings)
+        {
+            var effectiveSettings = settings ?? BigtableServiceApiSettings.GetDefault();
+            var apiConfig = new ApiConfig
+            {
+                ChannelPool = new ChannelPoolConfig
+                {
+                    MaxSize = (uint)effectiveSettings.MaxChannels,
```

Comment:
Would it be worth making the properties uint to avoid the cast?

Reply:
I was under the impression that public APIs should not use unsigned integers because they are not CLS-compliant. Is that no longer a concern?

Reply:
Any code generated from protobufs with uint values will expose that anyway. The CLS isn't nearly as important now as it used to be - and even VB have unsigned integer support now, I believe.

Reply:
Ok, updated